packages/next/src/client/components/router-reducer/reducers/navigate-reducer.ts

```diff
@@ -237,6 +237,12 @@ export function navigateReducer(
       }
 
       if (prefetchValues.aliased) {
+        // When alias is enabled, search param may not be included in the canonicalUrl.
+        // But we want to set url to canonicalUrl so that we use redirected path for fetching dynamic data.
+        if (canonicalUrlOverride) {
+          url.pathname = canonicalUrlOverride.pathname
```

Comment:
`url` is an input to this function; shouldn't mutate it. Better to reassign to a local variable.